wiki/dfg_nodes.md

```diff
@@ -0,0 +1,53 @@
+# Understanding Data Flow Graph (DFG) Node Labels in Flowr
+
+## Overview
+The Data Flow Graph (DFG) in Flowr represents the dependencies between different code components. Each node in the DFG contains specific labels that provide information about the operation being performed, the code span, and its relationship with other nodes.
+
+This document clarifies the meaning of these labels and how to interpret them.
+
+## Node Label Structure
+Each node in the DFG follows this general format:
+
+```
+[NodeType] (NodeID) LineRange ParentNodes
+```
+
+### **Breakdown of Components:**
+- **NodeType:** Indicates the type of operation. Example: `[RBinaryOp]` for a binary operation.
+- **NodeID:** A unique identifier for the node. Example: `(7)`.
```

Comment:
This id again links the vertex to its [Normalized AST](https://github.com/flowr-analysis/flowr/wiki/Normalized-AST). So yes, they are unique by design:

https://github.com/flowr-analysis/flowr/blob/7e7fd221dc521371cdae2dc08e009ffbbd749070/src/dataflow/graph/graph.ts#L158

Yet the brackets are no part of the id! In this case it is just the number.